x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/sync_integrations_on_remote.ts

```diff
@@ -180,3 +204,138 @@ export const syncIntegrationsOnRemote = async (
     }
   }
 };
+
+export const getFollowerIndexInfo = async (
+  esClient: ElasticsearchClient,
+  index: string,
+  logger: Logger
+) => {
+  if (!index) {
+    throw new IndexNotFoundError(`Follower index ${index} not found`);
+  }
+  try {
+    const res = await esClient.ccr.followInfo({
+      index,
+    });
+    if (!res?.follower_indices || res.follower_indices.length === 0)
+      throw new IndexNotFoundError(`Follower index ${index} not available`);
+
+    if (res.follower_indices[0]?.status === 'paused') {
+      throw new IndexNotFoundError(`Follower index ${index} paused`); // error code ?
+    }
+    return res.follower_indices[0];
+  } catch (err) {
+    if (err?.body?.error?.type === 'index_not_found_exception')
+      throw new IndexNotFoundError(`Index ${index} not found`);
+
+    logger.error('error', err.message);
+    throw err;
+  }
+};
+
+export const compareSyncedIntegrations = async (
+  esClient: ElasticsearchClient,
+  savedObjectsClient: SavedObjectsClientContract,
+  index: string,
+  logger: Logger
+) => {
+  try {
+
+    const searchRes = await esClient.search<SyncIntegrationsData>({
+      index,
+      size: 50, // ?
+      sort: [
+        {
+          'integrations.updated_at': {
+            order: 'desc',
+          },
+        },
+      ],
+    });
+    if (searchRes.hits.hits[0]?._source === undefined) {
+      return {
+        items: [],
+        error: `No integrations found on ${index}`
+      }
+    }
+    const ccrIntegrations = searchRes.hits.hits[0]?._source?.integrations;
+    const installedIntegrations = await getPackageSavedObjects(savedObjectsClient);
+    if (!installedIntegrations) {
+      return {
+        items: [],
+        error: `No integrations installed on remote`
```

Comment:
it's not necessarily an error if the main cluster doesn't have any integrations either